waspc/src/Wasp/Generator/WebAppGenerator/Auth/AuthFormsG.hs

```diff
@@ -0,0 +1,128 @@
+module Wasp.Generator.WebAppGenerator.Auth.AuthFormsG
+  ( genAuthForms,
+  )
+where
+
+import Data.Aeson (object, (.=))
+import StrongPath (reldir, relfile, (</>))
+import qualified Wasp.AppSpec.App.Auth as AS.Auth
+import Wasp.Generator.AuthProviders (gitHubAuthProvider, googleAuthProvider)
+import qualified Wasp.Generator.AuthProviders.OAuth as OAuth
+import Wasp.Generator.FileDraft (FileDraft)
+import Wasp.Generator.Monad (Generator)
+import Wasp.Generator.WebAppGenerator.Auth.Common (getOnAuthSucceededRedirectToOrDefault)
+import Wasp.Generator.WebAppGenerator.Common as C
+import Wasp.Util ((<++>))
+
+genAuthForms :: AS.Auth.Auth -> Generator [FileDraft]
+genAuthForms auth =
+  sequence
+    [ genAuthComponent auth,
+      genTypes auth,
+      copyTmplFile [relfile|auth/forms/Login.tsx|],
+      copyTmplFile [relfile|auth/forms/Signup.tsx|],
+      copyTmplFile [relfile|stitches.config.js|]
+    ]
+    <++> genEmailForms auth
+    <++> genInternalAuthComponents auth
+  where
+    copyTmplFile = return . C.mkSrcTmplFd
+
+genAuthComponent :: AS.Auth.Auth -> Generator FileDraft
+genAuthComponent auth =
+  return $
+    C.mkTmplFdWithData
+      [relfile|src/auth/forms/Auth.tsx|]
+      tmplData
+  where
+    tmplData = object ["isEmailAuthEnabled" .= AS.Auth.isEmailAuthEnabled auth]
+
+genTypes :: AS.Auth.Auth -> Generator FileDraft
+genTypes auth =
+  return $
+    C.mkTmplFdWithData
+      [relfile|src/auth/forms/types.ts|]
+      tmplData
+  where
+    tmplData = object ["isEmailAuthEnabled" .= AS.Auth.isEmailAuthEnabled auth]
+
+genEmailForms :: AS.Auth.Auth -> Generator [FileDraft]
+genEmailForms auth =
+  genConditionally isEmailAuthEnabled $
+    sequence
+      [ copyTmplFile [relfile|auth/forms/ResetPassword.tsx|],
+        copyTmplFile [relfile|auth/forms/ForgotPassword.tsx|],
+        copyTmplFile [relfile|auth/forms/VerifyEmail.tsx|]
+      ]
+  where
+    copyTmplFile = return . C.mkSrcTmplFd
+    isEmailAuthEnabled = AS.Auth.isEmailAuthEnabled auth
```

Comment:
```suggestion
genEmailForms :: AS.Auth.Auth -> Generator [FileDraft]
genEmailForms auth =
  genConditionally isEmailAuthEnabled $
    sequence
      [ genFileCopy [relfile|auth/forms/ResetPassword.tsx|],
        genFileCopy [relfile|auth/forms/ForgotPassword.tsx|],
        genFileCopy [relfile|auth/forms/VerifyEmail.tsx|]
      ]
  where
    genFileCopy = return . C.mkSrcTmplFd
    isEmailAuthEnabled = AS.Auth.isEmailAuthEnabled auth
```

To keep naming consistent with other generator files (for example https://github.com/wasp-lang/wasp/blob/main/waspc/src/Wasp/Generator/WebAppGenerator.hs#L76)